### PR TITLE
Ban creation of a database with user postgres

### PR DESCRIPTION
**Pull request by timohaas**

### CODE REVIEW

File: postgraas_server/management_resources.py [hunk 1]
```diff
@@ -137,6 +137,12 @@ def post(self):
         if not args['db_pwd']:
             abort(400, msg='The password may not be empty.')
 
+        if args['db_username'] == "postgres" or  "postgres@" in args['db_username']:
```
Comment (theMarix) [inline postgraas_server/management_resources.py]: Wouldn't it be easier to just check for `starts_with`? I am pretty sure the `@` would not be legal in any other position but I find it more explicit.

```suggestion
        if args['db_username'] == "postgres" or args['db_username'].startswith("postgres@"):
```

Also, I am wondering wether we do, and even should, allow `@` at all.

Comment (timohaas) [reply postgraas_server/management_resources.py]: Good idea, it also doesn't conflict with the other tests, which uses names like test_create_postgres_instance.

File: postgraas_server/management_resources.py [hunk 2]
```diff
@@ -137,6 +137,12 @@ def post(self):
         if not args['db_pwd']:
             abort(400, msg='The password may not be empty.')
 
+        if args['db_username'].startswith("postgres"):
+            abort(
+                406,
```
Comment (Fra-nk) [inline postgraas_server/management_resources.py]: I'm missing how this request (or the `db_username` parameter) relate to the **HTTP headers** `Accept`/`Accept-Charset`/...
At least that what I see status [406](https://tools.ietf.org/html/rfc7231#section-6.5.6) **should** be used for.

I'd suggest status [422](https://tools.ietf.org/html/rfc4918#section-11.2):
> [...] this error condition may occur if an XML request body contains well-formed (i.e., syntactically correct), but semantically erroneous, XML instructions.

File: postgraas_server/management_resources.py (hunk 1 above)
Comment (Fra-nk) [reply postgraas_server/management_resources.py]: * I suggest to be cautious here. IMHO `postgres_user` **should** be a valid value.
* https://www.postgresql.org/docs/11/auth-username-maps.html suggests that `@` by definition are valid in usernames.

Comment (theMarix) [reply postgraas_server/management_resources.py]: The former is why I only suggested that for the second part of the check.

I am aware that `@` is valid in usernames. This is required to have users that are host-specific. In our case we want to have users in a specific value for the host, though. Thus, it does not really make sense for the user to specify it.

Comment (Fra-nk) [reply postgraas_server/management_resources.py]: Ah, then I misunderstood you. Yeah, for `postgres@` your suggestion makes sense :).

**Pull request merged**